From 3e5c5260d3f2310f7ba4b40aa8a5497051f944ce Mon Sep 17 00:00:00 2001
From: Junwei Zhang <Jerry.Zhang@amd.com>
Date: Tue, 22 May 2018 16:26:41 +0800
Subject: [PATCH 4521/5725] drm/amdkcl: [4.17] fix prime bo for raven A+A issue

For kernel < 4.17, drm prime functions are not exported.
So retain the old path for prime bo sharing.

Change-Id: Icb4122ac0fcbada9a06006fbe31b9e80f0f1c9e9
Signed-off-by: Junwei Zhang <Jerry.Zhang@amd.com>
Reviewed-by: Le Ma <Le.Ma@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu.h     | 2 --
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 2 --
 2 files changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu.h b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
index 7ba0b6b..f683cb5 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
@@ -397,8 +397,6 @@ struct drm_gem_object *amdgpu_gem_prime_import(struct drm_device *dev,
                                                struct dma_buf *dma_buf);
 struct drm_gem_object *
 amdgpu_gem_prime_foreign_bo(struct amdgpu_device *adev, struct amdgpu_bo *bo);
-int amdgpu_gem_prime_pin(struct drm_gem_object *obj);
-void amdgpu_gem_prime_unpin(struct drm_gem_object *obj);
 struct reservation_object *amdgpu_gem_prime_res_obj(struct drm_gem_object *);
 void *amdgpu_gem_prime_vmap(struct drm_gem_object *obj);
 void amdgpu_gem_prime_vunmap(struct drm_gem_object *obj, void *vaddr);
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index 8222902..1c0cf9a 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -915,8 +915,6 @@ static struct drm_driver kms_driver = {
         .gem_prime_import = amdgpu_gem_prime_import,
 #else
 	.gem_prime_import = drm_gem_prime_import,
-        .gem_prime_pin = amdgpu_gem_prime_pin,
-        .gem_prime_unpin = amdgpu_gem_prime_unpin,
 #endif
 	.gem_prime_res_obj = amdgpu_gem_prime_res_obj,
 	.gem_prime_get_sg_table = amdgpu_gem_prime_get_sg_table,
-- 
2.7.4

